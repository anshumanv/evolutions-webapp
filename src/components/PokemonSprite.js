import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PokeSprite from 'react-poke-sprites';
import { BattlePokedex } from '../pokedex';

export default class PokemonSprite extends Component {
  static propTypes = {
    pokemon: PropTypes.string,
  };

  state = {
    hover: false,
  };

  onMouseEnter = event => {
    this.setState({ hover: true });
  };

  onMouseLeave = event => {
    this.setState({ hover: false });
  };

  render() {
    const { pokemon } = this.props;
    const { hover } = this.state;
    return (
      <span onMouseEnter={this.onMouseEnter} onMouseLeave={this.onMouseLeave}>
        <PokeSprite pokemon={pokemon} className="pokemons" alt={pokemon} />
        {hover && <div>{`${BattlePokedex[pokemon]}`}</div>}
      </span>
    );
  }
}
