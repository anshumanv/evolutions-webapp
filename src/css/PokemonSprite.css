.PokemonSprite-row {
  overflow: auto;
  height: 100%;
}

.PokemonSprite-col {
  float: left;
  width: 33%;
}

.PokemonSprite-container {
  text-align: left;
  width: 300px;
}
