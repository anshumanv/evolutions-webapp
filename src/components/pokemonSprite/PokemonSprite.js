import React from 'react';
import PropTypes from 'prop-types';
import PokeSprite from 'react-poke-sprites';
import 'react-tippy/dist/tippy.css';
import { Tooltip } from 'react-tippy';
import Info from './Info';

const PokemonSprite = ({ pokemon, info }) => (
  <Tooltip
    arrow
    interactive
    position="bottom"
    arrowSize="big"
    html={<Info info={info} />}
  >
    <PokeSprite pokemon={pokemon} className="pokemons" alt={pokemon} />
  </Tooltip>
);

PokemonSprite.propTypes = {
  pokemon: PropTypes.string,
  info: PropTypes.object,
};

export default PokemonSprite;
