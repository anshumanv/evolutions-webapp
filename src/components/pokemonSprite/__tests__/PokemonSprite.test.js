import React from 'react';
import { shallow } from 'enzyme';
import PokemonSprite from '../PokemonSprite';

describe('PokemonSprite', () => {
  const getWrapper = () => shallow(<PokemonSprite pokemon="pikachu" />);
  it('renders', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
