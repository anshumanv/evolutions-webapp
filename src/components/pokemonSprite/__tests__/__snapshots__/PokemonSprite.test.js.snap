// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonSprite renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonSprite
    pokemon="pikachu"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animateFill": true,
      "animation": "shift",
      "arrow": true,
      "arrowSize": "big",
      "children": <PokeSprite
        alt="pikachu"
        className="pokemons"
        pokemon="pikachu"
/>,
      "className": "",
      "delay": 0,
      "disabled": false,
      "distance": 10,
      "duration": 375,
      "followCursor": false,
      "hideDelay": 0,
      "hideDuration": 375,
      "hideOnClick": true,
      "html": <Info
        info={undefined}
/>,
      "inertia": false,
      "interactive": true,
      "interactiveBorder": 2,
      "multiple": false,
      "offset": 0,
      "onHidden": [Function],
      "onHide": [Function],
      "onRequestClose": [Function],
      "onShow": [Function],
      "onShown": [Function],
      "popperOptions": Object {},
      "position": "bottom",
      "size": "regular",
      "sticky": false,
      "stickyDuration": 200,
      "style": Object {},
      "theme": "dark",
      "touchHold": false,
      "trigger": "mouseenter focus",
      "unmountHTMLWhenHide": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alt": "pikachu",
        "className": "pokemons",
        "pokemon": "pikachu",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animateFill": true,
        "animation": "shift",
        "arrow": true,
        "arrowSize": "big",
        "children": <PokeSprite
          alt="pikachu"
          className="pokemons"
          pokemon="pikachu"
/>,
        "className": "",
        "delay": 0,
        "disabled": false,
        "distance": 10,
        "duration": 375,
        "followCursor": false,
        "hideDelay": 0,
        "hideDuration": 375,
        "hideOnClick": true,
        "html": <Info
          info={undefined}
/>,
        "inertia": false,
        "interactive": true,
        "interactiveBorder": 2,
        "multiple": false,
        "offset": 0,
        "onHidden": [Function],
        "onHide": [Function],
        "onRequestClose": [Function],
        "onShow": [Function],
        "onShown": [Function],
        "popperOptions": Object {},
        "position": "bottom",
        "size": "regular",
        "sticky": false,
        "stickyDuration": 200,
        "style": Object {},
        "theme": "dark",
        "touchHold": false,
        "trigger": "mouseenter focus",
        "unmountHTMLWhenHide": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "pikachu",
          "className": "pokemons",
          "pokemon": "pikachu",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
